README.md

```diff
@@ -9,270 +9,314 @@ Benefits
+   (--detach or -d) mode.
 
-```sh
-docker compose up -d
-```
+   ```sh
+   docker compose up -d
+   ```
 
-To shutdown the containers:
+   To shutdown the containers:
 
-```sh
-docker compose down
-```
+   ```sh
+   docker compose down
+   ```
 
 ## Developing nomad + plugins locally.
 
-This guide explains how to set up a streamlined development environment for nomad-lab and its plugins using uv workspaces.
+This guide explains how to set up a streamlined development environment for nomad-lab and its plugins using 
+[`uv` workspaces](https://docs.astral.sh/uv/concepts/workspaces/#workspaces).
 This approach eliminates the need for multiple pip install commands by leveraging a monorepo and a single installation step.
 
-In this example, we'll set up the development environment for a developer working on the following computational parsers:
-
-- nomad-parser-plugins-electronic
-- nomad-parser-plugins-atomistic
-- nomad-parser-plugins-workflow
-- nomad-parser-plugins-database
+In this example, we'll set up the development environment for a developer working 
+on the following plugins: `nomad-parser-plugins-electronic` and 
+`nomad-measurements`. The first plugin already comes as a dependency
+in this dev distribution. On the contrary, the second plugin is not listed as a
+dependency. In the following, we take a look at how to setup the environment in
+these two situations.
 
 ### Step-by-Step Setup
 
 1. Update submodules
 
-```bash
-git submodule update --init --recursive
-```
+   This loads the `nomad-lab` package which is already listed as a submodule.
+   ```bash
+   git submodule update --init --recursive
+   ```
 
 2. Add local plugins
 
-Add the plugin repositories to the packages/ directory, either as submodules.
+   Assuming that you already have a git repo for your plugins, add them to the 
+   `packages/` directory as submodules. In case, you are looking to create a plugin
+   repo from scratch, consider using our plugin template:
+   [nomad-plugin-template](https://github.com/FAIRmat-NFDI/nomad-plugin-template).
+
+   ```bash
+   git submodule add https://github.com/<package_name>.git packages/<package_name>
+   ```
+
+   Repeat for all the plugin packages you want to add and develop. For our example:
 
-```bash
-git submodule add https://github.com/package_name.git packages/package_name
-```
+   ```bash
+   git submodule add https://github.com/nomad-coe/electronic-parsers.git packages/nomad-parser-plugins-electronic
 
-For example:
+   git submodule add https://github.com/FAIRmat-NFDI/nomad-measurements.git packages/nomad-measurements
+   ```
 
-```bash
-git submodule add https://github.com/nomad-coe/atomistic-parsers.git packages/nomad-parser-plugins-atomistic
-```
 
-Repeat for all local dev packages (e.g., nomad-parser-plugins-electronic, nomad-parser-plugins-atomistic, etc.).
+3. Modify `pyproject.toml`
 
-2. Modify pyproject.toml
+   To ensure `uv` recognizes the 
```

Comment:
I wonder if this needs a bit more background on what a local plugin is, etc.

Reply:
How about: local plugin (a local copy of your plugin repository available in `packages/` directory)?